test/run-integration-tests.sh

```diff
@@ -116,4 +116,19 @@ else
     echo -e "\nTOTAL FAILURES: $FAILURES OF $COUNT"
 fi
 
+# If we have coverage files, merge them into a single report for upload
+if [ -n "${ARTIFACT_DIRECTORY}" ]; then
+    lcov_args=""
+
+    for info_file in "${ARTIFACT_DIRECTORY}"/*.coverage-info; do
+        if [ "${info_file}" = "${ARTIFACT_DIRECTORY}/*.coverage-info" ]; then
```

Comment:
Also, when we're at it, you can get around this `if` by using something like:

```bash
while read -r info_file; do
    lcov_args+=(--add-tracefile "${info_file}")
done < <(find "${ARTIFACT_DIRECTORY}" -maxdepth 1 -name "*.coverage-info")
```

But this is just a suggestion, both version should yield the same results.

Reply:
yeah that looks better